### DISCUSSION TITLE
Google Play failed to upload artefacts. All uploaded bundles must be signed. Please sign the bundle using jarsigner

### DISCUSSION THREAD

**Discussion by arungowda09** [codemagic-ci-cd/codemagic-docs#2743] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Other - please mention in your question

### Steps to reproduce

start new build with the project it completes all the steps  but failed to upload to the google play
In **Publishing**  Step
Publishing failed :|
Google Play failed to upload artefacts. All uploaded bundles must be signed. Please sign the bundle using jarsigner.: {
    "error": {
        "code": 403,
        "message": "All uploaded bundles must be signed. Please sign the bundle using jarsigner.",
        "status": "PERMISSION_DENIED"
    }
}

sign in keys are proper verified 

### Expected results

expected result to get build successful

### Actual results

Publishing failed :|
Google Play failed to upload artefacts. All uploaded bundles must be signed. Please sign the bundle using jarsigner.: {
    "error": {
        "code": 403,
        "message": "All uploaded bundles must be signed. Please sign the bundle using jarsigner.",
        "status": "PERMISSION_DENIED"
    }
}

### Build id (optional)

66e89356b07f700f7da8a031

**Comment by himesh-cm** [comment]
Hey @arungowda09, please check if you setup code-signing within your android project as shown in the docs [here](https://docs.codemagic.io/yaml-code-signing/signing-android/#signing-android-apps-using-gradle).

**Reply by himesh-cm** [reply]
Hey @arungowda09, i have deleted the comment, please do not share sensitive credentials here. i am looking into this and will get back to you.

**Reply by himesh-cm** [reply]
Hey @arungowda09, If you check the logs for the section `Set up key.properties, ` you should see the error information below.

Please modify the script as shown [here](https://docs.codemagic.io/yaml-code-signing/signing-android/#signing-android-apps-using-user-specified-keys), mainly the `storeFile`.

please make the required change and re-run the build.

**Reply by arungowda09** [reply]
**Changed file as for the info** 
same issue
Publishing failed :|
Google Play failed to upload artefacts. All uploaded bundles must be signed. Please sign the bundle using jarsigner.: {
    "error": {
        "code": 403,
        "message": "All uploaded bundles must be signed. Please sign the bundle using jarsigner.",
        "status": "PERMISSION_DENIED"
    }
}

**Reply by shrinidhi** [reply]
@himesh-cm any solution with this problem?

**Reply by himesh-cm** [reply, marked as answer]
Hey @arungowda09, @shrinidhi, 

Please switch to the below method.

1. Please make the following changes to your yaml. For more details, refer to the document [here](https://docs.codemagic.io/yaml-code-signing/signing-android/#referencing-keystores-in-codemagicyaml).

```
environment:
      android_signing:
        - cookie_keystore
```

2. Please set up the gradle settings as shown [here](https://docs.codemagic.io/yaml-code-signing/signing-android/#signing-android-apps-using-gradle) in the docs.
3. Remove the `Set up key.properties` step. It is not required if we set up signing via gradle.

Let me know, if you encounter any issues.

**Comment by himesh-cm** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
